gcs/src/gcs_group.cpp

```diff
@@ -1005,6 +1009,63 @@ group_recount_votes (gcs_group_t& group)
     return true;
 }
 
+/* Function to recompute vote based on only on the error code */
+int64_t recompute_vote_based_on_error_code (const gu::GTID& gtid,
+                                            std::string &err_msg,
+                                            int64_t old_vote,
+                                            bool needs_logging)
+{
+
+    // Define a regular expression to match either a 4-digit or a 5-digit error code following "Error_code:"
+    std::regex errCodeRegex(R"(Error_code: (\b\d{4}\b|\b\d{5}\b))");
+
+    // A vector to store the matched 4-digit and 5-digit error codes
+    std::vector<int> matchedErrorCodes;
+
+    // Use std::sregex_iterator to find all matches in the error message
+    std::sregex_iterator it(err_msg.begin(), err_msg.end(), errCodeRegex);
+    std::sregex_iterator end;
+
+    // Iterate through all matches and store them in the vector
+    while (it != end)
+    {
+        std::smatch match = *it;
+        int number = std::stoi(match[1].str()); // Extract and convert the error code to integer
```

Comment:
We are converting string to int here and a few lines below we do just the opposite. Maybe it is enough to keep them as strings?
Sorting will be different as it will sort lexicographically, but it should not matter as all nodes will sort in the same way.

Reply:
Done. Removed sorting.

Reply:
Why sorting was also removed? Is it somehow guaranteed that errors order is the same on all nodes?

Reply:
Sorting restored.